.github/workflows/release.yml

```
@@ -21,23 +47,26 @@ jobs:
           distribution: 'graalvm'
           github-token: ${{ secrets.GITHUB_TOKEN }}
           native-image-job-reports: 'true'
-      - name: Build native images with nativ maven plugin and extract in tar.gz
+      - name: Download documentation
+        uses: actions/download-artifact@v4
+        with:
+          name: docs
+          path: ./cli/target/package
+      - name: Build native image
         shell: bash
         run: |
-          maven_config="$(cat .mvn/maven.config)"
-          current_version="${maven_config/#*-Drevision=}"
-          current_version="${current_version/ */}"
           cd cli
-          mvn -B -ntp -Pnative -DskipTests=true package
-      - uses: actions/upload-artifact@v3
+          mvn -B -ntp -P${{ matrix.os }} '-P!build-documentation' -DskipTests=true package
```

I would not create a profile for each matrix OS. Your 4 profiles for this in the POM are 98% copy&paste.
Simply keep a single `native` profile and pass the assembly XML file name part as variable:
```suggestion
          mvn -B -ntp -Dideasy.assembly.id=${{ matrix.os }} -DskipTests=true package
```